Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -12,7 +12,7 @@ option(BUILD_SHARED_LIBS "Build shared l
 option(WITH_SHARED_LIBUV "Link to a shared libuv library instead of static linking" OFF)
 option(WITH_LUA_ENGINE "Lua engine" "Lua")
 
-set(WITH_LUA_ENGINE "Lua")
+set(WITH_LUA_ENGINE "LuaJIT")
 set(LUA_BUILD_TYPE "System")
 set(WITH_SHARED_LIBUV ON)
 
