Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -2,7 +2,8 @@ BUILD_DIR=build_dir
 TARGET=ubus-scriptd
 SOURCE=src/main.c src/lua_object.c src/script_object.c
 OBJECTS=$(patsubst %.c,%.o,$(SOURCE))
-CFLAGS+=-Wall -Werror -std=gnu99 
+CFLAGS+=-Wall -Werror -std=gnu99
+LUA_LDFLAGS=$(shell pkg-config --libs luajit)
 
 all: $(BUILD_DIR) $(BUILD_DIR)/$(TARGET)
 
@@ -10,7 +11,7 @@ $(BUILD_DIR):
 	mkdir -p $(BUILD_DIR)
 
 $(BUILD_DIR)/$(TARGET): $(addprefix $(BUILD_DIR)/,$(OBJECTS)) 
-	$(CC) -o $@ $^ -lpthread -lubus -lubox -lblobmsg_json -ldl -llua -lm
+	$(CC) -o $@ $^ -lpthread -lubus -lubox -lblobmsg_json -ldl $(LUA_LDFLAGS) -lm
 
 $(BUILD_DIR)/%.o: %.c $(wildcard src/*.h)
 	@mkdir -p $(dir $@)
