Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
 PROJECT(jsonpath C)
-ADD_DEFINITIONS(-Os -Wall -Werror --std=gnu99 -Wmissing-declarations -Wno-error=unused-variable -ffunction-sections -D_GNU_SOURCE)
+ADD_DEFINITIONS(-Os -Wall -Werror --std=gnu99 -Wno-error=unused-variable -ffunction-sections -D_GNU_SOURCE)
 
 SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "-Wl,--gc-sections")
 
@@ -24,22 +24,15 @@ IF(JSONC_FOUND)
 ENDIF()
 
 ADD_CUSTOM_COMMAND(
-  OUTPUT contrib/lemon
-  DEPENDS contrib/lemon.c contrib/lempar.c
-  COMMAND gcc -o contrib/lemon contrib/lemon.c
-  COMMENT "Generating lemon parser generator"
-)
-
-ADD_CUSTOM_COMMAND(
-  OUTPUT parser.c
-  DEPENDS parser.y contrib/lemon
-  COMMAND ./contrib/lemon parser.y
+  OUTPUT ${CMAKE_SOURCE_DIR}/parser.c ${CMAKE_SOURCE_DIR}/parser.h
+  DEPENDS ${CMAKE_SOURCE_DIR}/parser.y
+  COMMAND lemon ${CMAKE_SOURCE_DIR}/parser.y
   COMMENT "Generating parser.c"
 )
 
 SET_PROPERTY(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES "lemon;parser.h;parser.out")
 SET_SOURCE_FILES_PROPERTIES("parser.c" PROPERTIES GENERATED TRUE)
-ADD_EXECUTABLE(jsonpath main.c ast.c lexer.c parser.c matcher.c)
+ADD_EXECUTABLE(jsonpath main.c ast.c lexer.c parser.c matcher.c parser.y)
 TARGET_LINK_LIBRARIES(jsonpath ubox ${json})
 
 INSTALL(TARGETS jsonpath RUNTIME DESTINATION bin)
