Index: git/lua/CMakeLists.txt
===================================================================
--- git.orig/lua/CMakeLists.txt
+++ git/lua/CMakeLists.txt
@@ -8,7 +8,7 @@ IF(NOT LUA_CFLAGS)
 	FIND_PROGRAM(PKG_CONFIG pkg-config)
 	IF(PKG_CONFIG)
 		EXECUTE_PROCESS(
-			COMMAND pkg-config --silence-errors --cflags lua5.1
+			COMMAND pkg-config --silence-errors --cflags luajit
 			OUTPUT_VARIABLE LUA_CFLAGS
 			OUTPUT_STRIP_TRAILING_WHITESPACE
 		)
Index: git/lua/test.lua
===================================================================
--- git.orig/lua/test.lua
+++ git/lua/test.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/usr/bin/env luajit
 
 require "ubus"
 require "uloop"
Index: git/lua/test_client.lua
===================================================================
--- git.orig/lua/test_client.lua
+++ git/lua/test_client.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/usr/bin/env luajit
 
 require "ubus"
 require "uloop"
