Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -3,18 +3,19 @@ IWINFO_CFLAGS      = $(CFLAGS) -std=gnu9
 IWINFO_LDFLAGS     = -luci
 
 IWINFO_LIB         = libiwinfo.so
-IWINFO_LIB_LDFLAGS = $(LDFLAGS) -shared
+IWINFO_LIB_LDFLAGS = -Wl,--soname=libiwinfo.so.0 $(LDFLAGS) -shared -luci 
 IWINFO_LIB_OBJ     = iwinfo_utils.o iwinfo_wext.o iwinfo_wext_scan.o iwinfo_lib.o
 
-IWINFO_LUA         = iwinfo.so
+IWINFO_LUA         = luaiwinfo.so
 IWINFO_LUA_LDFLAGS = $(LDFLAGS) -shared -L. -liwinfo -llua
 IWINFO_LUA_OBJ     = iwinfo_lua.o
 
 IWINFO_CLI         = iwinfo
-IWINFO_CLI_LDFLAGS = $(LDFLAGS) -L. -liwinfo
+IWINFO_CLI_LDFLAGS = -L. -liwinfo $(LDFLAGS)
 IWINFO_CLI_OBJ     = iwinfo_cli.o
 
 
+
 ifneq ($(filter wl,$(IWINFO_BACKENDS)),)
 	IWINFO_CFLAGS  += -DUSE_WL
 	IWINFO_LIB_OBJ += iwinfo_wl.o
@@ -36,8 +37,10 @@ endif
 %.o: %.c
 	$(CC) $(IWINFO_CFLAGS) $(FPIC) -c -o $@ $<
 
-compile: clean $(IWINFO_LIB_OBJ) $(IWINFO_LUA_OBJ) $(IWINFO_CLI_OBJ)
-	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_LIB_LDFLAGS) -o $(IWINFO_LIB) $(IWINFO_LIB_OBJ)
+compile: $(IWINFO_LIB_OBJ) $(IWINFO_LUA_OBJ) $(IWINFO_CLI_OBJ)
+	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_LIB_LDFLAGS) -o $(IWINFO_LIB).0.0 $(IWINFO_LIB_OBJ)
+	ln -sf $(IWINFO_LIB).0.0 $(IWINFO_LIB)
+	ln -sf $(IWINFO_LIB).0.0 $(IWINFO_LIB).0
 	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_LUA_LDFLAGS) -o $(IWINFO_LUA) $(IWINFO_LUA_OBJ)
 	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_CLI_LDFLAGS) -o $(IWINFO_CLI) $(IWINFO_CLI_OBJ)
 
Index: git/include/iwinfo.h
===================================================================
--- git.orig/include/iwinfo.h
+++ git/include/iwinfo.h
@@ -181,10 +181,10 @@ struct iwinfo_ops {
 	void (*close)(void);
 };
 
-const char * iwinfo_type(const char *ifname);
-const struct iwinfo_ops * iwinfo_backend(const char *ifname);
-const struct iwinfo_ops * iwinfo_backend_by_name(const char *name);
-void iwinfo_finish(void);
+extern const char * iwinfo_type(const char *ifname);
+extern const struct iwinfo_ops * iwinfo_backend(const char *ifname);
+extern const struct iwinfo_ops * iwinfo_backend_by_name(const char *name);
+extern void iwinfo_finish(void);
 
 extern const struct iwinfo_ops wext_ops;
 extern const struct iwinfo_ops madwifi_ops;
